router/router.go

```diff
@@ -411,14 +415,46 @@ func (worker *workerT) workerProcess() {
 						WorkspaceId:   job.WorkspaceId,
 					}
 					worker.rt.responseQ <- jobResponseT{status: &status, worker: worker, userID: userID, JobT: job}
-					worker.rt.logger.Debugf(`Decrementing in throttle map for destination:%s since job:%d is marked as waiting for user:%s`, parameters.DestinationID, job.JobID, userID)
-					worker.rt.throttler.Dec(parameters.DestinationID, userID, 1, worker.throttledAtTime, throttler.ALL_LEVELS)
 					continue
 				}
 			}
 
-			firstAttemptedAt := gjson.GetBytes(job.LastJobStatus.ErrorResponse, "firstAttemptedAt").Str
+			var (
+				err            error
+				limited        bool
+				throttlingCost int64
+				tokensReturner throttling.TokenReturner
+				throttler      = worker.rt.getThrottler(parameters.DestinationID)
+			)
+			if len(worker.routerJobs) > 0 {
+				throttlingCost = worker.getThrottlingCost()
+			}
```

Comment:
It seems like we are calculating the throttling cost of all existing router jobs for every job, before adding this job to the router jobs. Why?